src/mpid/ch4/netmod/ofi/ofi_win.c

```diff
@@ -148,18 +148,21 @@ static int win_allgather(MPIR_Win * win, void *base, int disp_unit)
     MPIDI_OFI_WIN(win).mr = NULL;
     if (base || (win->create_flavor == MPI_WIN_FLAVOR_DYNAMIC && !MPIDI_OFI_ENABLE_MR_ALLOCATED)) {
         /* device buffers are not currently supported */
-        if (MPIR_GPU_query_pointer_is_dev(base))
+        if (MPIR_GPU_query_pointer_is_dev(base)) {
             rc = -1;
-        else
-            MPIDI_OFI_CALL_RETURN(fi_mr_reg(MPIDI_OFI_global.ctx[vni].domain,   /* In:  Domain Object */
-                                            base,       /* In:  Lower memory address */
-                                            win->size,  /* In:  Length              */
-                                            FI_REMOTE_READ | FI_REMOTE_WRITE,   /* In:  Expose MR for read  */
-                                            0ULL,       /* In:  offset(not used)    */
-                                            MPIDI_OFI_WIN(win).mr_key,  /* In:  requested key       */
-                                            0ULL,       /* In:  flags               */
-                                            &MPIDI_OFI_WIN(win).mr,     /* Out: memregion object    */
-                                            NULL), rc); /* In:  context             */
+        } else {
+            size_t len;
+            if (win->create_flavor == MPI_WIN_FLAVOR_DYNAMIC) {
+                len = UINTPTR_MAX - (uintptr_t) base;
+            } else {
+                len = (size_t) win->size;
+            }
+
+            MPIDI_OFI_CALL_RETURN(fi_mr_reg(MPIDI_OFI_global.ctx[vni].domain, base, len,
+                                            FI_REMOTE_READ | FI_REMOTE_WRITE, 0ULL,
+                                            MPIDI_OFI_WIN(win).mr_key, 0ULL,
+                                            &MPIDI_OFI_WIN(win).mr, NULL), rc);
+        }
```

Comment:
Can we keep the original comment for each parameter of `fi_mr_reg`? They are useful. 

Besides, it will be better if we can avoid unnecessary changes (e.g., trying to change only the line of `win->size,  /* In:  Length              */` in the `fi_mr_reg` block. This is a very "sensitive" function. We often use `git blame` to check the reason of updates. So, let's avoid unnecessary change

Reply:
Alright. It is unfortunate that we have very different views on these practices.